dev-haskell/spago/spago-0.20.3.ebuild

```diff
@@ -0,0 +1,81 @@
+HOMEPAGE="https://github.com/purescript/spago#readme"
+SRC_URI="https://github.com/purescript/${PN}/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/purescript/purescript-docs-search/releases/download/v0.0.10/purescript-docs-search -> ${PN}-purescript-docs-search-0.0.10
+	https://github.com/purescript/purescript-docs-search/releases/download/v0.0.10/docs-search-app.js -> ${PN}-docs-search-app-0.0.10.js
+	https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/purescript-docs-search -> ${PN}-purescript-docs-search-0.0.11
+	https://github.com/purescript/purescript-docs-search/releases/download/v0.0.11/docs-search-app.js -> ${PN}-docs-search-app-0.0.11.js
+	"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+
+# Does not compile with >=dev-haskell/versions-5
+RDEPEND="dev-haskell/aeson:=[profile?]
+	dev-haskell/aeson-pretty:=[profile?]
+	dev-haskell/ansi-terminal:=[profile?]
+	dev-haskell/async-pool:=[profile?]
+	dev-haskell/bower-json:=[profile?]
+	dev-haskell/cryptonite:=[profile?]
+	>=dev-haskell/dhall-1.38.0:=[profile?]
+	dev-haskell/either:=[profile?]
+	dev-haskell/file-embed:=[profile?]
+	dev-haskell/foldl:=[profile?]
+	dev-haskell/fsnotify:=[profile?]
+	dev-haskell/generic-lens:=[profile?]
+	dev-haskell/glob:=[profile?]
+	dev-haskell/http-client:=[profile?]
+	dev-haskell/http-conduit:=[profile?]
+	dev-haskell/http-types:=[profile?]
+	dev-haskell/lens-family-core:=[profile?]
+	dev-haskell/megaparsec:=[profile?]
+	dev-haskell/mtl:=[profile?]
+	dev-haskell/network-uri:=[profile?]
+	dev-haskell/open-browser:=[profile?]
+	dev-haskell/optparse-applicative:=[profile?]
+	dev-haskell/prettyprinter:=[profile?]
+	dev-haskell/retry:=[profile?]
+	>=dev-haskell/rio-0.1.16.0:=[profile?]
+	dev-haskell/rio-orphans:=[profile?]
+	dev-haskell/safe:=[profile?]
+	dev-haskell/semver-range:=[profile?]
+	dev-haskell/stm:=[profile?]
+	dev-haskell/stringsearch:=[profile?]
+	dev-haskell/tar:=[profile?]
+	dev-haskell/temporary:=[profile?]
+	<dev-haskell/text-1.3:=[profile?]
+	dev-haskell/turtle:=[profile?]
+	dev-haskell/unliftio:=[profile?]
+	dev-haskell/unordered-containers:=[profile?]
+	dev-haskell/utf8-string:=[profile?]
+	<dev-haskell/versions-5:=[profile?]
+	dev-haskell/with-utf8:=[profile?]
+	dev-haskell/zlib:=[profile?]
+	>=dev-lang/ghc-8.8.3:=
+"
+DEPEND="${RDEPEND}
+	>=dev-haskell/cabal-3.0.1.0"
+#	test? ( dev-haskell/extra
+#		>=dev-haskell/hspec-2 <dev-haskell/hspec-3
+#		dev-haskell/hspec-megaparsec
+#		dev-haskell/quickcheck )
+
+src_prepare() {
+	cp -v "${DISTDIR}/${PN}-purescript-docs-search-0.0.10" "${S}/templates/purescript-docs-search-0.0.10" || die
+	cp -v "${DISTDIR}/${PN}-docs-search-app-0.0.10.js" "${S}/templates/docs-search-app-0.0.10.js" || die
+	cp -v "${DISTDIR}/${PN}-purescript-docs-search-0.0.11" "${S}/templates/purescript-docs-search-0.0.11" || die
+	cp -v "${DISTDIR}/${PN}-docs-search-app-0.0.11.js" "${S}/templates/docs-search-app-0.0.11.js" || die
```

Comment:
Having to type out every file like this seems error prone and hard to update. Unfortunately, I couldn't figure out a programmatic way to do this, especially since `SRC_URI` doesn't seem to recognize any new variables I define in the ebuild. If anyone has any suggestions of how to do this better, let me know.

Reply:
These files are required to be in the `${S}/templates/` directory during compile time, and are usually downloaded by a script during `make`.

Reply:
Also: do I have the order right by having `default` at the end of `src_prepare`? I remember somebody saying that it should be at the end so that users can apply patches on top of the changes introduced in this function, but I see a lot of ebuilds putting `default` at the beginning.

Reply:
I'm not sure if there is a right answer for where to put `default`. Leaving it at the end means that your patches will always succeeded, because they're being applied to a known state.

Putting it at the beginning means that the user may patch it in such a way that your patches fail. Pick your poison i guess.

Regarding the files getting moved: first, i think it's great that you figured out where to get these from this is a big step forward for the purescript ebuild!

Second, there may be a better way to do it, I'll have to think on it when I'm back at my comp.